yt_dlp/extractor/goodgame.py

```diff
@@ -0,0 +1,46 @@
+from .common import InfoExtractor
+from ..utils import clean_html
+
+_GOODGAME_API = 'https://api2.goodgame.ru/v2'
+_GOODGAME_HLS = 'https://hls.goodgame.ru'
+
+
+class GoodGameIE(InfoExtractor):
+    IE_NAME = 'goodgame:stream'
+    _VALID_URL = r'https?://goodgame\.ru/channel/(?P<id>[a-zA-Z0-9_]+)'
+    _TESTS = [{
+        'url': 'https://goodgame.ru/channel/lol4to',
+        'md5': '14007abd9cd1d8589aed04f25ad15782',
+        'info_dict': {
+            'id': 'lol4to',
+            'ext': 'm3u8',
+            'title': 'Дота 1 на iCCup.com - Эффект Даннинга — Крюгера в вакууме.',
+            'description': 'Youtube stream\nСтрим на YouTube/Записи со стримов\nVk Music\n\nDonation Alerts',
+            'thumbnail': r're:^https?://.*\.jpg$',
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': True,
+        },
+        # 'skip': 'HTTP Error 404: Not Found',
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        gg = self._download_json(f'{_GOODGAME_API}/streams/{video_id}', video_id)
+        title = gg['channel']['title']
+        description = clean_html(gg['channel']['description'])
+        thumb = str(gg['channel']['thumb'])
+        if thumb.startswith('//'):
+            thumb = 'https://' + thumb.removeprefix('//')
+
+        gg_id = gg['id']
+        stream_url = f'{_GOODGAME_HLS}/manifest/{gg_id}_master.m3u8'
```

Comment:
read contributing guidelines:
1. Fields should be non-fatal
2. remove unnecessary variables

Reply:
1. Can you describe how to do it properly?
2. What vars are unnecessary here?

Reply:
Migrated to `traverse_obj`,
Removed `_GOODGAME_HLS` and `_GOODGAME_API` vars.